### PR TITLE
Improve code coverage for kubectl set

### PR DESCRIPTION
**Pull request by zjj2wry**
**What this PR does / why we need it**:
improve test cover from 69.7% to 75.3%

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:
think you ~

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubectl/cmd/set/set_resources_test.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+		},
+	}
+	for _, testCase := range testCases {
+		err := testCase.resourcesOptions.Validate()
+		if err != nil {
+			if err.Error() != testCase.expectErr {
+				t.Errorf("[%s]:expect err:%s got err:%s", testCase.name, testCase.expectErr, err.Error())
+			}
+		}
+		if err == nil && (testCase.expectErr != "") {
```
Comment (soltysh) [inline pkg/kubectl/cmd/set/set_resources_test.go]: Parentheses not needed.

File: pkg/kubectl/cmd/set/set_resources_test.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+			name: "test invalid argument",
+			resourcesOptions: &ResourcesOptions{
+				Requests: "cpu=500m memory=1Gi",
+			},
+			expectErr: "Invalid argument syntax cpu=500m memory=1Gi, expected <resource>=<value>",
+		},
+	}
+	for _, testCase := range testCases {
+		err := testCase.resourcesOptions.Validate()
+		if err != nil {
```
Comment (soltysh) [inline pkg/kubectl/cmd/set/set_resources_test.go]: Can you combine the two ifs together?

**Review by soltysh — COMMENTED** [review]
Small nits, otherwise looks good.

File: pkg/kubectl/cmd/set/set_resources_test.go (hunk 2 above)
Comment (zjj2wry) [inline pkg/kubectl/cmd/set/set_resources_test.go]: i attempt it, but **cmdutil.CheckErr(options.Validate())** will exit(code). i cannot get error to compare with my expectOut.
```
admindeMacBook-Pro:set admin$ go test
error: you must specify an update to requests or limits (in the form of --requests/--limits)
exit status 1
FAIL	k8s.io/kubernetes/pkg/kubectl/cmd/set	0.097s
```

**Comment by zjj2wry** [comment]
@soltysh has rebase

**Comment by zjj2wry** [comment]
@fabianofranz can you help review this? think you :)

**Comment by fabianofranz** [comment]
@k8s-bot ok to test

**Comment by zjj2wry** [comment]
@k8s-bot pull-kubernetes-e2e-gce-etcd3 test this

**Comment by fabianofranz** [comment]
/approve
I'll let @soltysh do the lgtm.
Thanks @zjj2wry

File: pkg/kubectl/cmd/set/set_resources_test.go (hunk 2 above)
Comment (soltysh) [inline pkg/kubectl/cmd/set/set_resources_test.go]: @zjj2wry sorry for my long silence, I was OOO for quite a while. Anyway, I've tried this:
```go
		if err != nil && err.Error() != testCase.expectErr {
			t.Errorf("[%s]:expect err:%s got err:%s", testCase.name, testCase.expectErr, err.Error())
		}
```
here and it works like expected. Mind updating?

**Review by soltysh — CHANGES_REQUESTED** [review]

**Comment by zjj2wry** [comment]
@soltysh ok, i will fix this. thank you reply 😊

**Comment by zjj2wry** [comment]
@soltysh ok, i will fix this. thank you 😊

**Pull request closed without merging**